lib/task.ml

```diff
@@ -34,12 +35,78 @@ let setup_pool ~num_additional_domains =
   let domains = Array.init num_additional_domains (fun _ -> Domain.spawn worker) in
   {domains; task_chan}
 
-let async pool task =
+let teardown_pool pool =
+  for _i=1 to Array.length pool.domains do
+    Multi_channel.send pool.task_chan Quit
+  done;
+  Multi_channel.clear_local_state ();
+  Array.iter Domain.join pool.domains
+
+ module Pool = struct
+
+    let table = Hashtbl.create 8
+
+    let pool_mutex = Mutex.create ()
+
+    let default = Atomic.make true
+
+    let is_default_active () = Atomic.get default
+
+    let default_pool = Atomic.make (setup_pool ~num_additional_domains:3)
```

Comment:
Currently the default pool is active by default. This makes it not mandatory to initialize a pool in the beginning of the application code.
There are some downsides to this; leaving idle domains typically increases GC overhead that could potentially slowdown I/O heavy programs. Alternatively, the default pool can be activated with an explicit call to `setup_default_pool`, if people are okay with the latter approach we can switch to that.